src/main/kotlin/widgets/Selector.kt

```diff
@@ -0,0 +1,178 @@
+                        private val colorAccent : Color,
+                        private val updateSelections : PublishSubject<DataSelectionInterface>,
+                        private val preferredSelection : PublishSubject<DataSelectionInterface>
+) : Fragment() {
+
+    init {
+        messages = ResourceBundle.getBundle("MyView")
+    }
+
+    private val compositeDisposable = CompositeDisposable()
+    private val viewModel = SelectorViewModel(updateSelections, preferredSelection)
+
+
+    private val chips = mutableListOf<Chip>()
+    private val selectionList = SelectionList(selectionData)
+    private val input = SimpleObjectProperty<String>()
+
+
+    private val colorNeutral = c(messages["UI_NEUTRAL"])
+    private val textFillNeutral = c(messages["UI_NEUTRALTEXT"])
+
+    override val root = vbox {
+
+        alignment = Pos.CENTER
+
+        label(label)
+
+        combobox(input, selectionList.observableList) {
+
+            // Color specific style options are handled in the widget
+            // This way a new css rule isn't needed for each color
+            style {
+                borderColor = multi(box(colorAccent))
+                focusColor = colorAccent
+            }
+
+            /**
+             * Allow filtered searching and filter based on the display string or
+             * any other strings affiliated with the data item.
+             */
+
+            isEditable = true
+            promptText = hint
+            makeAutocompletable(false) {
+                selectionList.dataList.filter {
+                    current -> current.filterText.contains(it)||
+                        current.labelText.contains(it, true)
+                }.map { it.labelText }.sorted()
+            }
+
+            /**
+             * Select any text in the textfield when it is refocused
+             */
+            editor.focusedProperty().addListener {
+                _, _, _ -> run {
+                    Platform.runLater {
+                        if (editor.isFocused && !editor.text.isEmpty()) {
+                            editor.selectAll()
+                        }
+                    }
+                }
+            }
+
+            /**
+             * Force the selected value of the combobox to be what is in the text editor.
+             * This will force the user to specify which item they want and remove cases
+             * where an item is accidentally selected
+             */
+            editor.textProperty().addListener { _, _, newText -> this.setValue(newText) }
+
+            /**
+             * Check and see if a valid item is selected whenever the combobox dropdown is closed.
+             * If so, then save the item and make a chip for it.
+             */
+            addEventFilter(ComboBox.ON_HIDDEN) {
+                if (selectionList.observableList.contains(input.value)) {
+                    viewModel.addNewValue(selectionList.dataList[selectionList.observableList.indexOf(input.value)])
```

Comment:
line seems pretty long

Reply:
Moved selectionList.observableList.indexOf(input.value) into a variable (index) and passed that into viewModel.addNewValue(selectionList.dataList[ index ]) on the next line